src/main/java/nl/hsac/fitnesse/junit/HsacFitNesseRunner.java

```diff
@@ -140,31 +143,48 @@ protected void runPages(List<WikiPage> pages, RunNotifier notifier) {
 
     }
 
-    // In the original runner class, we cannot use a system property as annotation value,
-    // so we override it to work exactly like the suiteFilter annotation.
-    // This also needs the ExcludeSuiteFilter interface
     @Override
-    protected String getExcludeSuiteFilter(Class<?> klass) throws Exception {
-        ExcludeSuiteFilter excludeSuiteFilterAnnotation = (ExcludeSuiteFilter)klass.getAnnotation(ExcludeSuiteFilter.class);
-        if (excludeSuiteFilterAnnotation == null) {
-            return null;
-        } else if (!"".equals(excludeSuiteFilterAnnotation.value())) {
-            return excludeSuiteFilterAnnotation.value();
-        } else if (!"".equals(excludeSuiteFilterAnnotation.systemProperty())) {
-            return System.getProperty(excludeSuiteFilterAnnotation.systemProperty());
+    protected boolean getSuiteFilterAndStrategy(Class<?> klass) throws Exception {
+        String strategy = System.getProperty(SUITE_FILTER_STRATEGY_OVERRIDE_VARIABLE_NAME);
+        if(StringUtils.isEmpty(strategy)){
+            return super.getSuiteFilterAndStrategy(klass);
+        } else if (strategy.equalsIgnoreCase("and")){
+            return true;
         } else {
-            throw new InitializationError("In annotation @ExcludeSuiteFilter you have to specify either \'value\' or \'systemProperty\'");
+            return false;
+        }
+    }
+
+    @Override
+    protected String getSuiteFilter(Class<?> klass) throws Exception {
+        String suiteFilter = System.getProperty(SUITE_FILTER_OVERRIDE_VARIABLE_NAME);
+        if (StringUtils.isEmpty(suiteFilter)) {
+            SuiteFilter suiteFilterAnnotation = klass.getAnnotation(SuiteFilter.class);
+            if (suiteFilterAnnotation == null) {
+                return null;
+            }
+            suiteFilter = suiteFilterAnnotation.value();
+        }
+        return suiteFilter;
+    }
+
+    @Override
+    protected String getExcludeSuiteFilter(Class<?> klass) throws Exception {
+        String excludeSuiteFilter = System.getProperty(EXCLUDE_SUITE_FILTER_OVERRIDE_VARIABLE_NAME);
+        if (StringUtils.isEmpty(excludeSuiteFilter)) {
+            ExcludeSuiteFilter excludeSuiteFilterAnnotation = klass.getAnnotation(ExcludeSuiteFilter.class);
```

Comment:
In this case, we can call the method from the superclass. (Simply null or value. No extra's there)